### ISSUE TITLE
Inference issue

### ISSUE THREAD

**Issue by NourSoltani** [VSainteuf/pytorch-psetae#15]
Greetings Mr @VSainteuf 
I have been trying to do inference using the pre-trained weights that you've provided, and on the  toy version of the Pixel-set dataset..
And it worked successfully like we can see below:

$ python inference.py --dataset_folder data --weight_dir /home/Desktop/pytorch-psetae/weights/CVPREPO_FINAL/
{'T': 1000,
 'batch_size': 128,
 'd_k': 32,
 'dataset_folder': 'data',
 'device': 'cuda',
 'dropout': 0.2,
 'fold': 'all',
 'geomfeat': 1,
 'input_dim': 10,
 'lms': None,
 'mlp1': [10, 32, 64],
 'mlp2': [132, 128],
 'mlp3': [512, 128, 128],
 'mlp4': [128, 64, 32, 20],
 'n_head': 4,
 'npixel': 64,
 'num_classes': 20,
 'num_workers': 8,
 'output_dir': './output',
 'pooling': 'mean_std',
 'positions': 'bespoke',
 'weight_dir': '/home/Desktop/pytorch-psetae/weights/CVPREPO_FINAL/'}
Preparation . . . 
Loading pre-trained models . . .
Successfully loaded 5 model instances
Inference . . .
100%|███████████████████████████████████████████████████████████████████████████████████| 4/4 [00:01<00:00,  2.52it/s]
Results stored in directory ./output

**But, I have 3 questions if you may:
First, the output file is named "Predictions_id_ytrue_y_pred.npy"   and its shape equals to (499,3) (You can find it attached).. So what does that mean? I mean what's the information that it provides?
Second, In order to do the inference, you should provide the META data which contains labels.json, dates.json ...etc. Isn't it?
So, in case I want to do the inference on a sentinel2 image of my own; from another tile of the globe and know its class.. Am I obliged to provide its META data? Because I thought that we should only provide the satellite image, and it will do the rest. Am I wrong?

Finally, I was wondering about how to prepare the "pkl"(mean_std) file?

Thank you!

[file.zip](https://github.com/VSainteuf/pytorch-psetae/files/10314814/file.zip)

**Comment by VSainteuf** [comment]
Hi @NourSoltani,
1. If you look [there](https://github.com/VSainteuf/pytorch-psetae/blob/fbd8ca3b5f6ebce6c53bfd1a0c25925dc2d7de52/inference.py#L60) you can see that Predictions_id_ytrue_y_pred.npy contains the id, the target,  and the prediction for each agricultural parcel of the test set. 
2. You are correct in the sense that we do not use the labels for training at inference. However, we do need the labels to measure the performance of the model through accuracy metrics. You can definitely do inference without labels and not measure performance, you will just have to slightly modify the code accordingly. 
3. for the normalisation file, you just need to compute the channel mean and std across your whole dataset (or just training dataset if you have a train/test split). The resulting tuple of tensors (mean, std) is stored in a pickle file. 
Hope this helps.
Cheers

**Comment by NourSoltani** [comment]
Thank you very much Mr @VSainteuf 
Yes.. It does help!  Thank you!
